src/parser/paladin/protection/modules/talents/Seraphim.tsx

```diff
@@ -37,47 +39,61 @@ class Seraphim extends Analyzer {
     if (!this.active) {
       return;
     }
-    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SERAPHIM_TALENT), this.onCast);
+    this.addEventListener(Events.applybuff.to(SELECTED_PLAYER).spell(SPELLS.SERAPHIM_TALENT), this.onSeraphimGain);
     const localStatTracker = (options.statTracker as StatTracker);
-    const bonusStats = this.getBonusStats(localStatTracker);
-    localStatTracker.add(SPELLS.SERAPHIM_TALENT.id, bonusStats);
-
+    localStatTracker.addStatMultiplierBuff(SPELLS.SERAPHIM_TALENT.id, this.statMultipliers);
   }
 
   get uptime(): number {
     return this.selectedCombatant.getBuffUptime(SPELLS.SERAPHIM_TALENT.id) / this.owner.fightDuration;
   }
 
   get averageMasteryGain(): number {
-    return this.bonusStatGains.map((gain) => gain.mastery).reduce((sum, next) => sum + next, 0) / this.bonusStatGains.length;
+    return STAT_MODIFIER * this.uptime;
   }
 
   get averageCritGain(): number {
-    return this.bonusStatGains.map((gain) => gain.crit).reduce((sum, next) => sum + next, 0) / this.bonusStatGains.length;
+    return STAT_MODIFIER * this.uptime;
   }
 
   get averageVersatilityGain(): number {
-    return this.bonusStatGains.map((gain) => gain.versatility).reduce((sum, next) => sum + next, 0) / this.bonusStatGains.length;
+    return STAT_MODIFIER * this.uptime;
   }
 
   get averageHasteGain(): number {
-    return this.bonusStatGains.map((gain) => gain.haste).reduce((sum, next) => sum + next, 0) / this.bonusStatGains.length;
+    return this.bonusStatGains.map((gain: BonusStats) => gain.haste).reduce((sum, next) => sum + next, 0) / this.bonusStatGains.length * this.uptime;
   }
 
-  onCast(event: CastEvent): void {
+  get statMultipliers() {
+    return {
+      haste: 1.0 + STAT_MODIFIER,
```

Comment:
Omit haste from the interactions with `StatTracker` entirely, since it uses Bloodlust-style addition which is covered by the `Haste` module. The `Haste` module listens for stat changes, so it will double-count haste if it shows up in both modules.